### PR TITLE
Drop unreliable run-time library loading

### PR DESCRIPTION
**Pull request by dg0yt**
Just one commit ATM.

However, there is more work left to do:
- the pkg-config file. It needs more Libs.private/Requires.private stuff.
- the cmake config. It needs more find_dependency.

Can you continue that part?

### CODE REVIEW

File: ports/sdl2-image/CMakeLists.txt [hunk 1]
```diff
@@ -3,27 +3,17 @@ project(SDL2_image C)
...
 # patch library names for preprocessor flags
 set(JPEG_FLAG JPG)
 set(TIFF_FLAG TIF)
-
-# names of potentially dynamically loaded libraries
-set(JPEG_DYNAMIC \"libjpeg-9${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}\")
-set(PNG_DYNAMIC \"libpng16-16${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}\")
-set(TIFF_DYNAMIC \"libtiff-5${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}\")
-set(WEBP_DYNAMIC \"libwebp-4${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}\")
+set(WebP_FLAGS WEBP)
```
Comment (Thomas1664) [inline ports/sdl2-image/CMakeLists.txt:16]:
```suggestion
set(WebP_FLAG WEBP)
```

Comment (dg0yt) [reply ports/sdl2-image/CMakeLists.txt:16]: :+1: Good catch.

**Comment by Thomas1664** [comment]
@dg0yt I want to merge your changes to `portfile.cmake#29` and `CMakelists.txt#6` if you resolved the merge conflicts. I did the rest very differently than you and we talked about the formatting in `vcpkg_cmake_configure`.

**Comment by dg0yt** [comment]
Just pick these lines manually. No credit needed.

**Pull request closed without merging**